tests/e2e/mod.rs

```diff
@@ -19,60 +20,68 @@ use tracing::{info, warn};
 
 mod dev;
 mod subgraph;
+mod supergraph;
 
 const GRAPHQL_TIMEOUT_DURATION: Duration = Duration::from_secs(30);
 
 #[derive(Debug, Deserialize)]
-struct ReducedSupergraphConfig {
+pub struct ReducedSupergraphConfig {
     subgraphs: HashMap<String, ReducedSubgraphConfig>,
 }
+
 #[derive(Debug, Deserialize)]
 struct ReducedSubgraphConfig {
     routing_url: String,
 }
 
 impl ReducedSupergraphConfig {
-    pub fn get_subgraph_urls(self) -> Vec<String> {
+    pub fn get_subgraph_urls(&self) -> Vec<String> {
         self.subgraphs
             .values()
             .map(|x| x.routing_url.clone())
             .collect()
     }
-}
 
-const RETAIL_SUPERGRAPH_SCHEMA_NAME: &'static str = "supergraph-config-dev.yaml";
+    pub fn get_subgraph_names(&self) -> Vec<String> {
+        self.subgraphs.keys().map(|name| name.clone()).collect()
+    }
+}
 
 #[fixture]
 #[once]
-fn run_subgraphs_retail_supergraph() -> TempDir {
+fn working_dir() -> TempDir {
+    // Clone the Git Repository that's needed to a temporary folder
+    TempDir::new().expect("Could not create temporary directory")
+}
```

Comment:
I'm not sure I like breaking this apart so much as this, would it be better to have the `clone_retail_supergraph_repo` return the `TempDir` instead?